### PR TITLE
SSM Cross Account auto account creation

### PR DESCRIPTION
**Pull request by ep-93**
Added ssm-cross-account-access role, hardcoded more than I should such as the trust policy, I can change that when it works.

The root or main account to run this still needs to be created, however that will be on one account only, so no need for delegate access changes,.

### CODE REVIEW

**Review by julialawrence — CHANGES_REQUESTED** [review]
Can we narrow down the permissions any?

File: terraform/environments/bootstrap/delegate-access/iam.tf [hunk 1]
```diff
@@ -21,6 +21,39 @@ module "cross-account-access" {
 
 }
 
+module "ssm-cross-account-access" {
+  source = "github.com/ministryofjustice/modernisation-platform-terraform-cross-account-access?ref=v2.3.0"
+  providers = {
+    aws = aws.workspace
+  }
+  account_id                  = "763252494486:role/AWS-SSM-AutomationAdministrationRole"
```
[the remark this answers is not in the document]
Comment (ep-93) [reply terraform/environments/bootstrap/delegate-access/iam.tf:29]: Yeah, sorry, building it up, I know it's hardcoded right now, and the account number wont change, but I shall make it less hardcodey

[the remark this answers is not in the document]
Comment (ep-93) [reply terraform/environments/bootstrap/delegate-access/iam.tf:29]: Ah, it adds root on the end regardless. 

                + AWS = "arn:aws:iam::763252494486:role/AWS-SSM-AutomationAdministrationRole:root"

File: terraform/environments/bootstrap/delegate-access/iam.tf [hunk 2]
```diff
@@ -21,6 +21,118 @@ module "cross-account-access" {
...
+
+data "aws_iam_policy_document" "trust-relationship-policy" {
+  statement {
+    sid     = ""
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::763252494486:role/AWS-SSM-AutomationAdministrationRole"]
```
Comment (davidkelliott) [inline terraform/environments/bootstrap/delegate-access/iam.tf:115]: Please could you remove the hard coded account number, thanks

**Review by davidkelliott — CHANGES_REQUESTED** [review]

**Review by ewastempel — DISMISSED** [review]
LGTM

**Review by ewastempel — APPROVED** [review]

**Review by davidkelliott — APPROVED** [review]

**Pull request merged**